test/Microsoft.DotNet.Cli.Utils.Tests/GivenAScriptCommandResolver.cs

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Text;
+using Xunit;
+using Moq;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.ProjectModel;
+using Microsoft.DotNet.Tools.Test.Utilities;
+using Microsoft.Extensions.PlatformAbstractions;
+using System.Threading;
+using FluentAssertions;
+using NuGet.Frameworks;
+
+namespace Microsoft.DotNet.Cli.Utils.Tests
+{
+    public class GivenAScriptCommandResolver
```

Comment:
Well, both for this and for the DefaultCommandresolver, what I would test is the actual code there. That is, that it has a certain list of resolvers (given your current implementation) and that it iterates through them until it finds a matching one.

I would then move all of these individual tests to tests for each individual resolver, instead of relying on testing them indirectly.